gateway/request.go

```diff
@@ -13,6 +14,11 @@ type HTTPRequestParser interface {
 	// GetQoSService returns the qos for the service matching an HTTP request.
 	GetQoSService(context.Context, *http.Request) (protocol.ServiceID, QoSService, error)
 
+	// GetFallbackURL returns the fallback URL to use in case
+	// no endpoints are available for the requested service.
+	// Returns the fallback URL and a boolean indicating whether a fallback URL is configured.
+	GetFallbackURL(context.Context, *http.Request) (*url.URL, bool)
```

Comment:
Similar general note re placement of fallback specification/validation/handling mechanisms: protocol/shannon package seems a better fit. I also recommend to avoid promoting the request parser to handle anything besides request parsing.